Framework/CurveFitting/src/Functions/ProcessBackground.cpp

```diff
@@ -1096,8 +1096,8 @@ size_t RemovePeaks::excludePeaks(vector<double> v_inX, vector<bool> &v_useX,
 
   // Count non-excluded region
   size_t count = 0;
-  for (size_t i = 0; i < v_useX.size(); ++i)
-    if (v_useX[i])
+  for (auto &&i : v_useX)
```

Comment:
I also wasn't sure had to lookup the answer.

http://stackoverflow.com/questions/15927033/what-is-the-correct-way-of-using-c11s-range-based-for

The problem is that std::vector template is specialized for bool, with an implementation that packs the bools to optimize space (each boolean value is stored in one bit, eight "boolean" bits in a byte).

Because of that (since it's not possible to return a reference to a single bit), vector<bool> uses a so called "proxy iterator" pattern. A "proxy iterator" is an iterator that, when dereferenced, does not yield an ordinary bool &, but instead returns (by value) a temporary object, which is a proxy class convertible to bool. (See also this question and related answers here on StackOverflow.)

To modify in place the elements of vector<bool>, a new kind of syntax (using auto&&) must be used:
```
for (auto&& x : v)
    x = !x;
```